packages/cozy-sharing/src/components/Recipient.jsx

```diff
@@ -75,142 +77,169 @@ export const UserAvatar = ({ url, size, ...rest }) => (
+              </ActionMenuItem>
+
+              <hr />
+              <ActionMenuItem
+                onClick={onRevokeClick}
+                left={<Icon icon="trash" color="var(--pomegranate)" />}
+              >
+                <Text className="u-pomegranate">
+                  {isOwner
+                    ? t(`${documentType}.share.revoke.title`)
+                    : t(`${documentType}.share.revokeSelf.title`)}
+                </Text>
+                <Caption>
+                  {isOwner
+                    ? t(`${documentType}.share.revoke.desc`)
+                    : t(`${documentType}.share.revokeSelf.desc`)}
+                </Caption>
+              </ActionMenuItem>
+            </ActionMenu>
+          )}
+        </>
+      )}
+    </div>
+  )
 }
 
-const StatusWithBreakpoints = withBreakpoints()(translate()(withClient(Status)))
+const Status = ({ status, isMe, instance }) => {
+  const { t } = useI18n()
+
+  const isError =
+    !isMe && ['error', 'unregistered', 'mail-not-sent'].includes(status)
+  const isReady = isMe || status === 'ready'
 
-const Recipient = (props, { client, t }) => {
+  let text, icon
+  if (isReady) {
+    text = instance
+    icon = 'to-the-cloud'
+  } else if (isError) {
+    text = t('Share.status.error')
+    icon = 'warning'
+  } else {
+    text = ['pending', 'seen'].includes(status)
+      ? t(`Share.status.${status}`)
+      : t('Share.status.pending')
+
+    icon = status === 'seen' ? 'eye' : 'paperplane'
+  }
+
+  return (
+    <Media>
+      <Img className={styles['recipient-status-icon']}>
+        <Icon icon={icon} size={10} />
+      </Img>
+      <Bd>
+        <Caption className={classNames({ 'u-pomegranate': isError })}>
+          {text}
+        </Caption>
+      </Bd>
+    </Media>
+  )
+}
+
+const Recipient = props => {
+  const { t } = useI18n()
+  const client = useClient()
   const { instance, isOwner, status, ...rest } = props
   const isMe =
     (isOwner && status === 'owner') || instance === client.options.uri
   const defaultDisplayName = t(DEFAULT_DISPLAY_NAME)
   const defaultInitials = defaultDisplayName[0].toUpperCase()
   const name = Contact.getDisplayName(rest, defaultDisplayName)
+
   return (
-    <div className={classNames(styles['recipient'], 'u-mt-1')}>
-      <Avatar
-        text={Contact.getInitials(rest, defaultInitials)}
-        size={'small-plus'}
-        textId={name}
-      />
-      <div className={styles['recipient-ident-status']}>
-        <Identity
-          name={isMe ? t('Share.recipients.you') : name}
-          details={instance}
+    <CompositeRow
+      dense
+      className={classNames(styles['recipient'], 'u-ph-0')}
+      primaryText={
+        <Text className="u-ellipsis">
+          {isMe ? t('Share.recipients.you') : name}
+        </Text>
+      }
+      secondaryText={<Status status={status} isMe={isMe} instance={instance} />}
+      image={
+        <Avatar
+          text={Contact.getInitials(rest, defaultInitials)}
```

Comment:
Don't we use the new getInitial and displayName methods? At least the new displayName (the one @JF-Cozy is working on?)

Reply:
Actually no, the format of the contacts inthe sharings, especially before they get accepted, is not the one expected by cozy-client. We would need to support more formats in cozy-client and this PR has been going on for a long time, I don't want to add this on top now.